assets/javascripts/discourse/components/category-tag-chat-link.js

```diff
@@ -0,0 +1,52 @@
+import Component from "@ember/component";
+import { ajax } from "discourse/lib/ajax";
+import { action } from "@ember/object";
+import { schedule } from "@ember/runloop";
+
+export default Component.extend({
+  category: null,
+  tag: null,
+  show: false,
+
+  init() {
```

Comment:
I have no idea what we are _actually_ moving towards, but I thought we were supposed to use `@on("init")` for init functions and other ember things like willDestroyElement now?

https://github.com/discourse/discourse/blob/7ce927eaa84bfaa9e72bd2f371625f2f95f6c42e/app/assets/javascripts/discourse/app/models/category.js#L17-L17

It lets you avoid doing `this._super`

Reply:
I would be surprised... that line linked is 4 years old and I've done this style for every PR of chat that @eviltrout has been reviewing. I'm thinking it's the reverse, actually.

Reply:
So weird cause I swear Joffrey has told me the opposite. No worries I will just ask about it internally.

Reply:
I will rarely reject a PR for using them but I prefer not using decorators for this. See our internal discussion.